packages/ckeditor5-html-support/tests/htmlcomment.js

```diff
@@ -222,6 +223,263 @@ describe( 'HtmlComment', () => {
+			} );
+
+			expect( editor.getData() ).to.equal( '<p>Foo</p><!--foo-->' );
+		} );
+	} );
+
+	describe( 'removeHtmlComment()', () => {
+		it( 'should allow removing a comment with the given comment ID', () => {
+			editor.setData( '<p>Foo</p><p>Bar</p><p>Baz</p>' );
+
+			const firstCommentID = editor.model.change( writer => {
+				const root = editor.model.document.getRoot();
+				const position = writer.createPositionAt( root, 1 );
+
+				return htmlCommentPlugin.createHtmlComment( position, 'foo' );
+			} );
+
+			const secondCommentID = editor.model.change( writer => {
+				const root = editor.model.document.getRoot();
+				const position = writer.createPositionAt( root, 1 );
+
+				return htmlCommentPlugin.createHtmlComment( position, 'bar' );
+			} );
+
+			htmlCommentPlugin.removeHtmlComment( firstCommentID );
+
+			expect( editor.getData() ).to.equal( '<p>Foo</p><!--bar--><p>Bar</p><p>Baz</p>' );
+
+			htmlCommentPlugin.removeHtmlComment( secondCommentID );
+
+			expect( editor.getData() ).to.equal( '<p>Foo</p><p>Bar</p><p>Baz</p>' );
+		} );
+
+		it( 'should throw an error when a comment with the given comment ID does not exist', () => {
+			editor.setData( '<p>Foo</p><p>Bar</p><p>Baz</p>' );
+
+			editor.model.change( writer => {
+				const root = editor.model.document.getRoot();
+				const position = writer.createPositionAt( root, 1 );
+
+				return htmlCommentPlugin.createHtmlComment( position, 'bar' );
+			} );
+
+			expectToThrowCKEditorError( () => {
+				htmlCommentPlugin.removeHtmlComment( 'invalid-comment-id' );
+			}, /^html-comment-does-not-exist/, null );
+		} );
+	} );
+
+	describe( 'getHtmlCommentsInRange()', () => {
+		it( 'should return all comment marker IDs present in the specified range', () => {
+			editor.setData( '<p>Foo</p><p>Bar</p><p>Baz</p>' );
+
+			htmlCommentPlugin.createHtmlComment(
+				createPositionFromPath( [ 1, 0 ] ),
+				'foo'
+			);
+
+			htmlCommentPlugin.createHtmlComment(
+				createPositionFromPath( [ 2 ] ),
+				'bar'
+			);
+
+			const id3 = htmlCommentPlugin.createHtmlComment(
+				createPositionFromPath( [ 2, 1 ] ),
+				'foo'
+			);
+
+			const id4 = htmlCommentPlugin.createHtmlComment(
+				createPositionFromPath( [ 2, 3 ] ),
+				'foo'
+			);
+
+			const range = editor.model.change( writer => {
+				return writer.createRangeIn( root.getChild( 2 ) );
+			} );
+
+			expect( htmlCommentPlugin.getHtmlCommentsInRange( range ) ).to.deep.equal( [ id3, id4 ] );
+		} );
+
+		it( 'should return all comment marker IDs present in the specified range including comments at range boundaries', () => {
+			editor.setData( '<p>Foo</p><p>Bar</p><p>Baz</p>' );
+
+			htmlCommentPlugin.createHtmlComment(
+				createPositionFromPath( [ 1, 0 ] ),
+				'foo'
+			);
+
+			htmlCommentPlugin.createHtmlComment(
+				createPositionFromPath( [ 2 ] ),
+				'bar'
+			);
+
+			const posStart = createPositionFromPath( [ 2, 1 ] );
+			const posEnd = createPositionFromPath( [ 2, 3 ] );
```

Comment:
The `createPositionFromPath()` is not needed since this functionality is available directly in the model.

```suggestion
			const posStart = model.createPositionFromPath( [ 2, 1 ] );
			const posEnd = model.createPositionFromPath( [ 2, 3 ] );
```